src/memory/AmdFlash.hh

```diff
@@ -1,7 +1,10 @@
 #ifndef AMDFLASH_HH
 #define AMDFLASH_HH
 
+#include "DeviceConfig.hh"
```

Comment:
Very much a detail: after adding this #include you can remove the forward declaration on line 23.